### PR TITLE
New Transfer Page (ETH + Token)

### PR DESCRIPTION
**Pull request by jubalm**
Updated transfer page design

Screenshot 2023-02-28 at 1 10 22 PM

This PR replaces/removes `SendEthPage` component which can only do one type of transfer and combines the functionality in a single page

This also resolves comments on #74 

### TODO
- [ ] enable dynamic eth and token balance

### CODE REVIEW

File: app/ts/library/constants.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	return asset.type === 'native'
+}
+
+export function getTokenByChainId(chainId: string) {
+	return assets.filter(isTokenAsset).filter(asset => asset.chainId === chainId)
+}
+
+export function getNativeCurrencyByChainId(chainId: string) {
+	const match = assets.filter(asset => asset.chainId === chainId).find(isNativeAsset)
+	if (match === undefined) throw new Error(`Could not find the native currency for ${chainId}`)
```
Comment (KillariDev) [inline app/ts/library/constants.ts:120]: here we should just call it ether and allow user to transact even if we don't directly support the network

**Review by KillariDev — APPROVED** [review]
scroll bars appear on me:
image

on small screen I get this menu and you have to select item and then click outside to close it:
image

It always says "send weth" even if I have other tokens selected:
image

I feel we should not show decimals here, its not important information:
image

I have to clear everything to be able to send again, can we allow user to click send again even if they have rejected it and adjust values? I don't feel we need "clear form" button at all here. It seems you can edit the transaction and then the clear form button disapears and you can send again, which feels quite confusing
image

When I don't have Metamask or interceptor, I cannot see the token list. We could just default to Mainnet if there's no wallet, and allow user to play around even if the don't have wallet to find out what the application does.
image

When showing addresses, always use the checksummed address, you can get it with `ethers.utils.getAddress('0x4c9bbfc1fbd93dfb509e718400978fbeedf590e9')`
image

Copy address button doesn't do anything:
image

File: app/ts/library/constants.ts (hunk 1 above)
Comment (jubalm) [reply app/ts/library/constants.ts]: adjusted handling on https://github.com/DarkFlorist/web3-app/pull/75/commits/247a1732e18df1cd659ca40a8e0c757db7ea9adf

File: app/ts/components/TransferPage/AssetListing.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+		// update list on network change
+		tokens.value = getTokenByChainId(chainIdHex)
+		native.value = getNativeCurrencyByChainId(chainIdHex)
+	}
+
+	useSignalEffect(listenForNetworkChanges)
+
+	// TODO: handle unknowns
+	if (tokens.value === undefined || native.value === undefined) return null
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/TransferPage/AssetListing.tsx:30]: `getNetwork` is async but usually happens really quick that it's very hard to notice in the UI. still would be good to handle later with busy indicators

File: app/ts/components/TransferPage/ActiveAsset.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+				<div style={{ gridArea: 'title' }}>
+					<div class='font-bold'>
+						{metadata.name} <span class='text-white/50'>{metadata.symbol}</span>
+					</div>
+				</div>
+				<div class='text-sm leading-tight' style={{ gridArea: 'address' }}>
+					<span>{metadata.address}</span>
+				</div>
+				<div class='text-sm text-white/50 leading-tight' style={{ gridArea: 'description' }}>
+					<span>Token Contract with {metadata.decimals} decimals</span>
```
[the remark this answers is not in the document]
Comment (jubalm) [reply app/ts/components/TransferPage/ActiveAsset.tsx]: this should be resolved with the change sets from https://github.com/DarkFlorist/web3-app/pull/75/commits/57180d335a4379c01cf11dc9044a3cdc685f6d1c

**Comment by jubalm** [comment]
> scroll bars appear on me: image
> 
> on small screen I get this menu and you have to select item and then click outside to close it: image
> 
> It always says "send weth" even if I have other tokens selected: image
> 
> I feel we should not show decimals here, its not important information: image
> 
> I have to clear everything to be able to send again, can we allow user to click send again even if they have rejected it and adjust values? I don't feel we need "clear form" button at all here. It seems you can edit the transaction and then the clear form button disapears and you can send again, which feels quite confusing image
> 
> When I don't have Metamask or interceptor, I cannot see the token list. We could just default to Mainnet if there's no wallet, and allow user to play around even if the don't have wallet to find out what the application does. image
> 
> When showing addresses, always use the checksummed address, you can get it with `ethers.utils.getAddress('0x4c9bbfc1fbd93dfb509e718400978fbeedf590e9')` image
> 
> Copy address button doesn't do anything: image

I have moved the to create a separate issue for the remaining 3 items on this comment to #80

**Pull request merged**